sprint-3/step-2

mellonis opened the pull request.

mellonis committed:
Refactoring

- gets rid of default exports

mellonis committed:
Refactoring

- removes yarn.lock

mellonis committed:
Adds routing

- adds pages stubs
- adds routing
- MenuItem. adds onClick prop

mellonis committed:
Prepares page layout for rendering content in the center of the screen

mellonis committed:
Refactors

- enhance the page css class use

mellonis committed:
Adds form elements on pages

mellonis committed:
Adds lexemes for the forms

mellonis committed:
Adds Form component

mellonis committed:
Fixes the button rendering

mellonis committed:
Fixes the button rendering

mellonis committed:
Adds buttons on the forms

mellonis committed:
Fixes the additional actions rendering

mellonis committed:
Adds the produceAdditionalActionReactNode helper

mellonis committed:
Adds additional actions on the forms

mellonis committed:
Refactors

- simplifies the Form component usage

mellonis committed:
Adds input fields on the forms

- refactors the Form component
- adds inputs fields
- adds validation

mellonis committed:
Refactors

- renames reducer
- changes reducers export strategy

Refactors

- renames reducer

Refactors

- renames reducer

Refactors

- renames reducer

mellonis committed:
Protects route, adds redirects, implement auth interaction with server

mellonis committed:
Update README.md

mellonis committed:
#10 Profile Page. Reorganizes profile page

mellonis committed:
Refactors

- Forms
  - Adds support of initial values
  - Adds isButtonHiddenOnNotModifiedForm prop to hide submit button if there is no changes in the form
  - Adds resetButtonTitle prop to render a form reset button
  - Changes the form layout

mellonis committed:
#6 Profile. Render user data inputs on the profile page

mellonis committed:
Refactors

- Form
  - Adds showErrors prop to show form validation errors

mellonis committed:
Refactors

- Enable render form validation errors on profile, register and reset-password forms

mellonis committed:
#6 Profile. Adds ability to update user data

mellonis committed:
#7 Ingredients. Renders ingredient at direct link, changes route when ingredient modal is opened

mellonis committed:
#9 BurgerConstructor. Adds a check for user is logged in while place order workflow

mellonis committed:
ProfilePage. Adds the orders subpage

mellonis committed:
Login Page. Redirect to redirectedFrom page after successful authentication

mellonis committed:
Adds redirect to login page after logout

mellonis committed:
#8 Adds auto login functionality

mellonis committed:
Adds propTypes

mellonis committed:
Adds universal-cookie dependency

mellonis committed:
Adds changes according to checklist

mellonis committed:
Fixes the reset password functionality bug

mellonis committed:
Fixes the path unneeded path modification bug

mellonis committed:
Fixes the opened modal on history travels bug

gennady-bars commented on src/components/app-header/index.tsx:
```diff
@@ -7,11 +8,16 @@ import {
 } from '@ya.praktikum/react-developer-burger-ui-components';
 import { lexemes } from '../../consts';
-import MenuItem from './menu-item';
+import { MenuItem } from './menu-item';
```
при наведении на меню в хэдере лучше показывать курсор в виде руки `pointer`

gennady-bars commented on src/pages/main/index.tsx:
```diff
@@ -0,0 +1,97 @@
+      window.history.replaceState(
+        null,
+        '',
+        `/ingredients/${detailedIngredient._id}`
```
Вам нужно использовать средства роутинга для определения, показывать ли попап с деталями, или показывать страницу. Вам должны были дать ссылку на документацию по этой логике. Вот она https://reactrouter.com/web/example/modal-gallery . В ней используются `location` и `background`, которые передаются в `Switch`. Пожалуйста, посоветуйтесь с наставником.

gennady-bars reviewed (commented):
Здравствуйте. Работа проделана огромная. Роутинг очень хороший. Страницы и формы отличные. Даже валидация есть.


Но есть некоторые недочеты:

* при наведении на меню в хэдере лучше показывать курсор в виде руки `pointer`
* Вам нужно использовать средства роутинга для определения, показывать ли попап с деталями, или показывать страницу. Вам должны были дать ссылку на документацию по этой логике. Вот она https://reactrouter.com/web/example/modal-gallery . В ней используются `location` и `background`, которые передаются в `Switch`. Пожалуйста, посоветуйтесь с наставником.

**Можно лучше**

* при клике на логотип обычно идет переход на главную страницу
* желательно очищать конструктор после удачного получения номера заказа с сервера, чтобы пользователь мог следующий заказ сделать, не удаляя старые ингредиенты


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта перед отправкой на ревью. И лучше проверять наличие ошибок в консоли браузера перед отправкой (кнопка `F12`)

Удачного рефакторинга кода.

mellonis committed:
Refactors

- AppHeader. Updates style
- AppHeader. Adds a link from the logo to the main page

mellonis committed:
Business logic. Do clear the construction after placing an order

mellonis committed:
Bind the ingredient modal with the same path as the ingredient's page

mellonis committed:
Bind the ingredient modal with the same path as the ingredient's page

mellonis commented:
@gennady-bars , если вы имеете в виду сетевые ошибки, то их скрыть не возможно,
<img width="781" alt="Screenshot 2021-08-05 at 22 13 24" src="https://user-images.githubusercontent.com/1478873/128407906-44ec7fe6-1ca6-4317-81eb-685fa767a613.png">
 кроме как в инструмента разработчика

mellonis commented:
- поправил курсор для элементов шапки
- логотип ведёт на главную
- для отображения деталей ингредиента используются средства роутинга
- после размещения заказа конструктор очищается
- ошибок в консоли не замечено, кроме разве что сетевых, которые невозможно скрыть с помощью JS. Во всяком случае у меня не получилось, даже с помощью try catch (у меня запрос в async функции). Но, как писал ранее, их вывод можно предотвратить настройками панели разработчика.

gennady-bars commented:
> @gennady-bars , если вы имеете в виду сетевые ошибки, то их скрыть не возможно,
> <img alt="Screenshot 2021-08-05 at 22 13 24" width="781" src="https://user-images.githubusercontent.com/1478873/128407906-44ec7fe6-1ca6-4317-81eb-685fa767a613.png">
> кроме как в инструмента разработчика

про ошибки в консоли я написал на всякий случай, чтобы перед отправкой на ревью проверить, все ли в порядке с кодом. Многие этого не делают и не видят ошибок типизации, атрибута `key` и тд. 

У Вас было все отлично с этим

gennady-bars commented on src/components/app-body/index.tsx:
```diff
@@ -13,47 +14,81 @@ import {
+          <NotFoundPage />
+        </Route>
+      </Switch>
+      {backgroundPageLocation ? (
```
теперь отлично и универсально сделано.

В следующем спринте будут другие попапы,  которые теперь по этой универсальной схеме будут показывать страницы

gennady-bars commented on src/components/app-body/index.tsx:
```diff
@@ -13,47 +14,81 @@ import {
-);
+const AppBody = () => {
+  let location = useLocation();
+  const { state: { backgroundPageLocation } = {} } = useLocation() as {
```
на старте приложение ломается с ошибкой. Скрин http://joxi.ru/vAW8b9ZTB1pRPr

mellonis replied:
почему может быть так что у меня такой ошибки нет?

mellonis replied:
```ts 
  const { state: locationState } = useLocation() as {
    state: { backgroundPageLocation?: typeof location } | null;
  };
  const { backgroundPageLocation } = locationState ?? {};
```

если state может быть `null`, я могу переписать так.

что делать в этой ситуации? отправлять на ревью повторно?

gennady-bars reviewed (commented):
Придется отклонить без потери итерации, так как на старте приложение ломается с ошибкой. Скрин http://joxi.ru/vAW8b9ZTB1pRPr

mellonis committed:
Fixes a reported bug from the reviewer

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
